app/renderer/actions/Session.js

```diff
@@ -210,12 +211,24 @@ export function newSession (caps, attachSessId = null) {
           desiredCapabilities.testobject_api_key = session.server.testobject.apiKey || process.env.TESTOBJECT_API_KEY;
         }
         break;
-      case ServerTypes.headspin:
-        host = session.server.headspin.hostname;
-        port = session.server.headspin.port;
-        path = `/v0/${session.server.headspin.apiKey}/wd/hub`;
-        https = true;
+      case ServerTypes.headspin: {
+        if (!/^(http|https):\/\/(localhost|.+\.headspin.io)(|:)[0-9]*\/v\d+\/\w+\/wd\/hub$/.test(
```

Comment:
regexp matching is potentially error-prone. Why not to try url parsing first and then verify the parsed properties?